tools/debug/autodediffer.py

```diff
@@ -12,6 +12,12 @@
 from __future__ import print_function
 import os, sys
 
+from pathlib import Path
+
+sys.path.insert(1, str(Path(__file__).parents[2].resolve()))
+ 
+from tools import shared
```

Comment:
I agree I don't think its worth added a new dependency on shared to get these utils.   Just use Path.read here maybe?